backends-velox/src/main/scala/org/apache/gluten/metrics/TaskStatsAccumulator.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten.metrics
+
+import org.apache.spark.util.AccumulatorV2
+
+class TaskStatsAccumulator extends AccumulatorV2[String, String] {
+  private var stats: String = ""
+
+  override def isZero: Boolean = stats.isEmpty
+
+  override def copy(): AccumulatorV2[String, String] = {
+    val newAcc = new TaskStatsAccumulator()
+    newAcc.stats = this.stats
+    newAcc
+  }
+
+  override def reset(): Unit = {
+    stats = ""
+  }
+
+  override def add(v: String): Unit = {
+    if (stats.isEmpty) {
+      stats = v
+    } else {
+      stats += "\n" + v
```

Comment:
Is this going to be too long? Do we need to keep all the TaskStats?

Reply:
We can collect only long tasks, like more than 10s

Reply:
I'm still worried about long strings causing jvm oom, maybe we can limit the maximum length.